S3/Config.py

```diff
@@ -318,7 +318,16 @@ def role_config(self):
                     '%s=%s' % (k, s3_quote(v, unicode_output=True))
                     for k, v in params.items()
                 ])
-                conn = httplib.HTTPSConnection(host='sts.amazonaws.com',
+                sts_endpoint = "sts.amazonaws.com"
+                if os.environ.get("AWS_STS_REGIONAL_ENDPOINTS") == "regional": 
+                    # Check if the AWS_REGION variable is available to use as a region. 
+                    if os.environ.get("AWS_REGION"): 
+                        region = os.environ.get("AWS_REGION")
+                    else : 
+                        # Otherwise use the bucket location
+                        region = self.bucket_location
+                    sts_endpoint = f"sts.{region}.amazonaws.com"
```

Comment:
Can you use %-based string instead of f-string for retro compatibility?

Reply:
Hello, this has been fixed. 
I used the following : 
```python 
sts_endpoint = "sts.%s.amazonaws.com" % region
```

Let me know if this is how you wanted it.